kanidmd/src/lib/credential/mod.rs

```diff
@@ -706,6 +768,59 @@ impl Credential {
         }
     }
 
+    pub(crate) fn update_backup_code(&self, backup_code: HashSet<String>) -> Self {
+        let type_ = match &self.type_ {
+            // TODO: we should panic/reject CredentialType != PasswordMfa?
+            CredentialType::Password(pw) | CredentialType::GeneratedPassword(pw) => {
+                debug_assert!(false);
+                CredentialType::PasswordMfa(
```

Comment:
I don't think this is valid to do this convert, since a backup code exists to recover from lost webauthn/totp.